internal/instance/resource_instance.go

```diff
@@ -116,6 +117,21 @@ func (r InstanceResource) Schema(_ context.Context, _ resource.SchemaRequest, re
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
 				},
+        Validators: []validator.String{
+          stringvalidator.ConflictsWith(
+            path.Expressions{
+              path.MatchRoot("source_instance"), 
+              path.MatchRoot("source_file"),
+            }...
+          ),
+          stringvalidator.AtLeastOneOf(
```

Comment:
The current implementation enforces `atMostOneOf`, which is not the same as `AtLeastOneOf`.